torch/csrc/jit/codegen/cuda/partial_split_map.cpp

```diff
@@ -0,0 +1,87 @@
+#include <torch/csrc/jit/codegen/cuda/ir_iostream.h>
+#include <torch/csrc/jit/codegen/cuda/ir_utils.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_ir.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_ir_builder.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_ir_printer.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
+#include <torch/csrc/jit/codegen/cuda/partial_split_map.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+void PartialSplitMap::build(Fusion* fusion) {
+  const auto gpu_lower = GpuLower::current();
+  auto used_vals = fusion->usedMathVals();
+
+  for (auto tv : ir_utils::filterByType<TensorView>(used_vals)) {
+    auto exprs = ExprSort::getExprs(
+        fusion, {tv->domain()->domain().begin(), tv->domain()->domain().end()});
+    for (auto split : ir_utils::filterByType<Split>(exprs)) {
+      // Only needs to check root domains as partial split is only
+      // allowed with root domains
+      if (std::find(
```

Comment:
That's also fine. Non-root domains actually have zero in the fields, and it should be safe to add mapping entries for non-root domains. It's just not necessary and doesn't make sense since non-root domains are not supposed to have those properties.

Reply:
I just thought it would be a faster check than `std::find(...`

Reply:
That's probably true, but I'd feel something's wrong if we needed to worry about a performance cost of checking a domain is a root domain or not since it's such a fundamental concept.